scripts/build.sh

```diff
@@ -297,11 +307,37 @@ create_poudriere_ports()
 			exit_err "Failed extracting poudriere ports"
 		fi
 
+		#Source the ports-interactions scripts
+		. "$(dirname $0)/ports-interactions.sh"
```

Comment:
Not sure you have to do the 'dirname $0' here. The rest of the script is intentionally written to never 'cd' and it assumes it is started from the top-level directory. So in this case you could run './scripts/ports-interactions.sh'.

Additionally, it's probably best to just source this file a single time up at the top of the build.sh script instead of elsewhere in the file.

Reply:
Done. Now that script is sourced once at the top of build.sh.
I left the dirname $0 thing in because it is a more robust method of ensuring we can source the file. I have run into a few relative/absolute path issues in the builds before, so that just ensures it always does the right thing no matter how the build.sh script was started.